Source/DataSources/StaticGeometryPerMaterialBatch.js

```diff
@@ -168,16 +168,20 @@ define([
                 var entity = updater.entity;
                 var instance = this.geometry.get(entity.id);
 
-                attributes = this.attributes.get(instance.id.id);
-                if (!defined(attributes)) {
-                    attributes = primitive.getGeometryInstanceAttributes(instance.id);
-                    this.attributes.set(instance.id.id, attributes);
-                }
+                if(defined(instance)) {
```

Comment:
I'm not sure I understand how this can every be false.  I believe an instance should always be defined at this point.  If that's not the case, there are a bunch of other spots in the code that need to be updated (and we should be adding unit tests for this behavior).